dit/dit.py

```diff
@@ -176,26 +176,29 @@ def _trace_selection(self, group, subgroup, task):
     # ===========================================
     # Paths and files names
 
-    def _base_path(self):
-        path = os.path.expanduser(os.path.join(self.directory))
+    def _make_base_path(self, directory):
```

Comment:
This method is both "making" and "setting" the base_path. Thus, its name is not appropriate. My suggestion is to use this method to just make the base path and return it. Then, you set the base_path on line 811, like this:
`self.base_path = self._make_base_path(directory)`